app/javascript/components/widget-transition-controls.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useReducer } from 'react';
+import PropTypes from 'prop-types';
+import { mathMod } from 'ramda';
+
+const prefetchDelay = 1500;
+let noPrefetchRequestedYet = true;
```

Comment:
can we keep this in state? Would be nice to keep all of our side effects in one place.